pkg/kubelet/images/image_gc_manager.go

```diff
@@ -320,6 +321,10 @@ func (im *realImageGCManager) freeSpace(bytesToFree int64, freeTime time.Time) (
 	// Get all images in eviction order.
 	images := make([]evictionInfo, 0, len(im.imageRecords))
 	for image, record := range im.imageRecords {
+		if isImageUsed(image, imagesInUse) {
+			glog.V(5).Infof("Image ID %s is being used", image)
+			continue
+		}
```

Comment:
what was the impact of this earlier for a running container ? Did this cause the container to be killed if its image was GC'd ? If yes , why was it not detected until now ? Is the kubelet GC feature not enabled by default ?

Reply:
> Did this cause the container to be killed if its image was GC'd? If yes , why was it not detected until now ? 

This image gc is to delete unused images to free some space. No containers will be killed since deleting a image that is referred by a running container is forbid.

> Is the kubelet GC feature not enabled by default ?

Image GC is enabled  by default.

Reply:
So IIUC, this behavior was previously enforced by the runtime, but now you're making the Kubelet enforce it?

Reply:
@tallclair  When `kubelet` is trying to free space by deleting docker images, images that are used by running containers are not skipped.

Even if we let runtime to handle this, these images should not be deleted.

What I am trying to do here is to skip these in-use images to avoid unnecessary overheads and spam log messages as well.